synapse/handlers/pagination.py

```diff
@@ -72,13 +73,123 @@ def __init__(self, hs):
         self.auth = hs.get_auth()
         self.store = hs.get_datastore()
         self.clock = hs.get_clock()
+        self.config = hs.config
 
         self.pagination_lock = ReadWriteLock()
         self._purges_in_progress_by_room = set()
         # map from purge id to PurgeStatus
         self._purges_by_id = {}
         self._event_serializer = hs.get_event_client_serializer()
 
+        if self.config.retention_enabled:
+            # Run the purge jobs described in the configuration file.
+            for job in self.config.retention_purge_jobs:
+                self.clock.looping_call(
+                    run_as_background_process,
+                    job["interval"],
+                    "purge_history_for_rooms_in_range",
+                    self.purge_history_for_rooms_in_range,
+                    job["shortest_max_lifetime"], job["longest_max_lifetime"],
```

Comment:
`black` would shout at you for this on the develop branch:

```suggestion
                    job["shortest_max_lifetime"], 
                    job["longest_max_lifetime"],
```